docs/install/Knative-with-Kourier.md

```diff
@@ -0,0 +1,161 @@
+bash in a Mac or Linux environment.
+
+## Install Knative
+
+Let's do a core install of Knative Serving with the released yaml templates:
+
+1.  To install Knative, first install the CRDs by running the following `kubectl apply`
+    command. This prevents race conditions during the install, which cause intermittent errors:
+
+        kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving-crds.yaml
+
+2.  To complete the install of Knative and its dependencies, next run the
+    following `kubectl apply` command:
+
+        kubectl apply --filename https://github.com/knative/serving/releases/download/{{< version >}}/serving-core.yaml
+
+3.  Monitor the Knative Serving namespace and wait until all of the pods come up with a
+    `STATUS` of `Running`:
+
+    ```
+    kubectl get pods -w -n knative-serving
+    ```
+
+
+## Install Kourier
+
+Knative default installation uses Istio to handle internal and external traffic. If you are just interested in exposing Knative applications to the external network, a service mesh adds overhead and increases the system complexity. Kourier provides a way to expose your Knative application in a more simple and lightweight way.
+
+You can install Kourier with `kubectl`:
+
+```
+kubectl apply \
+  --filename https://raw.githubusercontent.com/3scale/kourier/master/deploy/kourier-knative.yaml
+```
+
+## Configuring the Knative ingress class
+
+Kourier only exposes ingresses that have the "kourier" ingress class. By default Knative annotates all the ingresses for Istio but you can change that by patching the "config-network" configmap as follows: 
+
+```
+kubectl patch configmap/config-network \
+  -n knative-serving \
+  --type merge \
+  -p '{"data":{"ingress.class":"kourier.ingress.networking.knative.dev"}}'
+```
+
+## Configuring DNS
+
+Installing Kourier will create a Kubernetes Service with type `LoadBalancer`.
+This may take some time to get an IP address assigned, during this process, it
+will appear as `<pending>`.  You must wait for this IP address to be assigned
+before DNS may be set up.
+
+Get this external IP address with:
+
+```
+$ kubectl get svc kourier -n knative-serving
+
+NAME         TYPE           CLUSTER-IP     EXTERNAL-IP     PORT(S)        AGE
+kourier      LoadBalancer   10.43.242.100  172.22.0.2      80:31828/TCP   19m
+
+```
+
+This external IP can be used with your DNS provider with a wildcard `A` record;
+however, for a basic functioning DNS setup (not suitable for production!) this
+external IP address can be added to the `config-domain` ConfigMap in
+`knative-serving`. You can edit this with the following command:
+
+```
+kubectl edit cm config-domain --namespace knative-serving
+```
+
+Given the external IP above, change the content to:
```

Comment:
You'll need to add a line break after "to:"; all code blocks need to have white space before and after.